File: src/main/java/com/sogeti/codingchallenge/day05/model/Seat.java
package com.sogeti.codingchallenge.day05.model;

public class Seat {

    private final String boardingPass;
    private final int row;
    private final int col;

    public Seat(String boardingPass, int row, int col) {
        this.boardingPass = boardingPass;
        this.row = row;
        this.col = col;
    }

    public int getSeatID(int numberOfColumns) {
        return numberOfColumns * row + col;
    }

    @Override
    public String toString() {
        return "Seat{" +
                "boardingPass='" + boardingPass + '\'' +
                ", row=" + row +
                ", col=" + col +
                '}';
    }
}

File: src/main/java/hzt/aoc/day05/model/Seat.kt
package hzt.aoc.day05.model

class Seat(private val boardingPass: String, private val row: Int, private val col: Int) {
    fun getSeatID(numberOfColumns: Int): Int {
        return numberOfColumns * row + col
    }

    override fun toString(): String {
        return "Seat{" +
                "boardingPass='" + boardingPass + '\'' +
                ", row=" + row +
                ", col=" + col +
                '}'
    }
}
